core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -472,6 +475,7 @@ public ManagedChannelImpl shutdownNow() {
     List<OobTransportProviderImpl> oobTransportsCopy;
     synchronized (lock) {
       transportsCopy = new ArrayList<TransportSet>(transports.values());
+      transportsCopy.addAll(decommissionedTransports);
```

Comment:
Yeah, I agree. Although really, it is just scary more than anything. I've not gotten them confused at all (other than saved\* or \*Copy versions).